distributed/stealing.py

```diff
@@ -148,7 +148,7 @@ def log(self, msg):
     def add_worker(self, scheduler=None, worker=None):
         self.stealable[worker] = tuple(set() for _ in range(15))
 
-    def remove_worker(self, scheduler=None, worker=None):
+    def remove_worker(self, scheduler: Scheduler, worker: str):
         del self.stealable[worker]
```

Comment:
FWIW I thought about modifying / tracking removed workers and match up with in_flight. That led me towards a "remember cancellation" mechanism which was way too complicated.

Reply:
Okay. I'm a little hesitant about not clearing out the state properly, but I guess it's ok.

Reply:
See also https://github.com/dask/distributed/issues/6600 I would like to clarify the discussion on that ticket before we implement any more complex logic for work stealing